src/Package/Impl/ProjectSystem/Commands/SetDirectoryHereCommand.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.VisualStudio.ProjectSystem.Designers;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.R.Package.Commands;
+using Microsoft.VisualStudio.R.Package.Repl.Commands;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.R.Packages.R;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.Commands {
+    [ExportCommandGroup("AD87578C-B324-44DC-A12A-B01A6ED5C6E3")]
+    [AppliesTo("RTools")]
+    [OrderPrecedence(200)]
+    internal sealed class SetDirectoryHereCommand : ICommandGroupHandler {
+        public CommandStatusResult GetCommandStatus(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, string commandText, CommandStatus progressiveStatus) {
+            if (commandId == RPackageCommandId.icmdSetDirectoryHere && nodes.IsSingleNodePath()) {
+                return new CommandStatusResult(true, commandText, CommandStatus.Enabled | CommandStatus.Supported);
+            }
+            return CommandStatusResult.Unhandled;
+        }
+
+        public bool TryHandleCommand(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut) {
+            if (commandId == RPackageCommandId.icmdSetDirectoryHere) {
+                var path = nodes.GetNodeFolderPath();
+                if (!string.IsNullOrEmpty(path)) {
+                    var o = new object();
+
+                    var interactiveWorkflowProvider = VsAppShell.Current.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
+                    var interactiveWorkflow = interactiveWorkflowProvider.GetOrCreate();
+                    var controller = ReplCommandController.FromTextView(interactiveWorkflow.ActiveWindow.TextView);
+
+                    controller.Invoke(RGuidList.RCmdSetGuid, RPackageCommandId.icmdSetWorkingDirectory, path, ref o);
```

Comment:
Not a part of this change, but it would be great to wire up these `ICommandGroupHandler` and `IAsyncCommandGroupHandler` interfaces to our controller-based invocation in general.